pdc/apps/auth/views.py

```diff
@@ -159,8 +158,15 @@ def get_users_and_groups(arg):
         if user.is_superuser:
             users_list.append(str(user.username))
         else:
-            resource_permission_set = get_resource_permission_set(user)
-            if arg in resource_permission_set:
+            if read_permission_for_all():
+                resource_permission_set = set([obj for obj in models.ResourcePermission.objects.filter(
+                    permission__name__iexact='read')])
+            group_id_list = [group.id for group in user.groups.all()]
+            queryset = models.GroupResourcePermission.objects.filter(group__id__in=group_id_list)
+
+            for group_resource_permission in queryset:
+                resource_permission_set.add(group_resource_permission.resource_permission)
+            if resource_permission in resource_permission_set:
```

Comment:
Seems don't have too much different. Still low efficiency and lot of duplicate code of `get_resource_permission_set`. Actual logic part also has some duplication. Already got group list have this resource permission before, here check user -> groups -> resource permissions again.

And again, Should we consider a non-super user get permissions out of his group now?

It would be nice to respond to every comment and question. No matter just something like `fixed` or some other discussion.